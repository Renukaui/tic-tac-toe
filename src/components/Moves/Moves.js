import React from "react";

const Moves = props => {
	const { history, stepNumber, ...other } = props;
	const moves = history.map((step,move) => {
		let desc = move ? 'Move to Step #'+move+' ' : 'Go to game start';
		const btn_highlight = (stepNumber === move)?'btn-primary':'btn-secondary';
		return(
			<li key={move}>
	          	<button className={"btn "+btn_highlight+" btn-block"} 
	            onClick={()=> other.onClick(move) }>{desc}</button>
	        </li>
		);
	});
	
	return(
		<div className="game-info__moves">
			<ol className="list-moves list-unstyled">
				{moves}
			</ol>
		</div>
	);
}

export default Moves;