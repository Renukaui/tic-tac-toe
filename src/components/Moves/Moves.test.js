import React from 'react';
import Moves from './Moves';
import Adapter from 'enzyme-adapter-react-16';
import {shallow, configure} from 'enzyme';

configure({adapter: new Adapter()});

describe("Moves Component", () => {
  let props;
  beforeEach(() => {
    props = {
      history: [{ 
          squares:Array(9).fill(null), 
          clickIndex:1
      }],
      onClick: jest.fn()
    }
  })

  it('should match snapshot', () => {
    const wrapper = shallow(<Moves {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render without crashing', () => {
    shallow(<Moves {...props}/>);
  });
  it('should have go to start game button', () => {
    const wrapper = shallow(<Moves {...props}/>);
    const startGameBtn = wrapper.find('button.btn-block').text();
    expect(startGameBtn).toEqual('Go to game start');
  });
  it('should restart game when clicking button start game', () => {
    const wrapper = shallow(<Moves {...props}/>);
    const startGameBtn = wrapper.find('button.btn-block');
    startGameBtn.simulate('click');
    expect(props.onClick).toBeCalledWith(0)
  });
});