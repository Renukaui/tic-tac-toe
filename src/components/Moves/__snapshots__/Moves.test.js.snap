// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Moves Component should match snapshot 1`] = `ShallowWrapper {}`;
